frontend/packages/core-auth/src/components/AuthWrapper.tsx

```diff
@@ -8,8 +8,21 @@ import {
   AuthWrapperProps,
   CodeChallengeMethod,
   ResponseType,
+  TokenResponseConfig,
 } from '../types/types';
 import { TokenResponse } from '../types/response';
+import { getSessionStorage } from '../utils/getSessionStorage';
+
+const createTokenResponse = (trc: TokenResponseConfig) => {
```

Comment:
I'd probably move this to a static method on `TokenResponse`. Like `fromJSON`. It also doesn't need to use a `let`.

```
  if (!trc) return undefined;
  try {
    return new TokenResponse(trc);
  } catch (e) {
    return undefined;
  }
```